src/main/java/org/broadinstitute/consent/http/resources/DaaResource.java

```diff
@@ -83,6 +87,46 @@ public Response createDaaForDac(
     }
   }
 
+  @PUT
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed({ADMIN, SIGNINGOFFICIAL})
+  @Path("{daaId}/{userId}")
+  public Response createLibraryCardDaaRelation(
+      @Context UriInfo info,
+      @Auth AuthUser authUser,
+      @PathParam("daaId") Integer daaId,
+      @PathParam("userId") Integer userId) {
+    try {
+      User authedUser = userService.findUserByEmail(authUser.getEmail());
+      int authedUserInstitutionId = authedUser.getInstitutionId();
+      User user = userService.findUserById(userId);
+      int userInstitutionId = user.getInstitutionId();
+      // Assert that the user has the correct institution permissions to add a DAA-LC relationship.
+      // Admins can add a DAA with any DAC, but signing officials can only create relationships for
+      // library cards associated with the same institution they are associated with.
+      if (!authedUser.hasUserRole(UserRoles.ADMIN)) {
+        if (authedUserInstitutionId != userInstitutionId) {
+          return Response.status(Status.FORBIDDEN).build();
+        }
+      }
```

Comment:
I think this might be more clearly expressed as:

```suggestion
      if (authedUser.hasUserRole(UserRoles.SIGNINGOFFICIAL) && authedUserInstitutionId != userInstitutionId) {
          return Response.status(Status.FORBIDDEN).build();
      }
```

The non-admin role could be researcher etc, but the endpoint is limited to admins and signing officials.